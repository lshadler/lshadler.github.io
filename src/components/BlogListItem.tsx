import * as React from 'react';
import styled from '@emotion/styled';

import { Link } from 'gatsby';
import { breakpoints } from '../styles/variables';
import { getEmSize } from '../styles/mixins';
import { BlogPostData } from '../types.ts/index';

interface BlogListItemProps {
    data: BlogPostData
}

const ListItemContainer = styled.li`
    min-height: ${getEmSize(3)}em;
    border: ${getEmSize(1)}em solid var(--textTitle);
    margin: 1em;

`;

const StyledLink = styled(Link)`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    padding: 1em;

    @media (max-width: ${breakpoints.md}px) {
      flex-direction: column-reverse;
    }
`;

const StyledTitle = styled.div``;

const StyledDate = styled.div`
`;

const BlogListItem: React.FC<BlogListItemProps> = ({
  data,
}) => {
  const { frontmatter: { title, date }, fields: { slug } } = data;
  const dateObj: Date = new Date(date);

  return (
    <ListItemContainer>
      <StyledLink to={slug}>
        <StyledTitle>{title}</StyledTitle>
        <StyledDate>{dateObj.toLocaleDateString()}</StyledDate>
      </StyledLink>
    </ListItemContainer>
  );
};

export default BlogListItem;
