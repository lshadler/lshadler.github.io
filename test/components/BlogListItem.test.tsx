import React from 'react';
import { render } from '@testing-library/react';

import BlogListItem from '../../src/components/BlogListItem';
import { BlogPostData } from '../../src/types.ts';

describe('BlogListItem', () => {
  let data: BlogPostData;

  beforeEach(() => {
    data = {
      fields: {
        slug: 'blog-post-one',
      },
      frontmatter: {
        title: 'Blog Post One',
        date: '1995/11/11',
      },
    };
  });

  it('renders correctly', () => {
    const { asFragment } = render(<BlogListItem data={data} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
