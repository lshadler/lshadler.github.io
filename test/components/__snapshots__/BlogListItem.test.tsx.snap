// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogListItem renders correctly 1`] = `
<DocumentFragment>
  <li />
  <li
    class="css-1tse97z"
  >
    <a
      class="css-1aqrz0m"
      href="blog-post-one"
    >
      <div
        class="css-0"
      >
        Blog Post One
      </div>
      <div
        class="css-fpqub2"
      >
        11/11/1995
      </div>
    </a>
  </li>
</DocumentFragment>
`;
